scripts/script-tag.js

```diff
@@ -0,0 +1,418 @@
+    
+            if (time.length > 1) { // If time format correct
+                time = time.slice(1); // Remove full string match value
+                time[5] = +time[0] < 12 ? 'am' : 'pm'; // Set AM/PM
+                time[0] = +time[0] % 12 || 12; // Adjust hours
+            }
+            return time.join(''); // return adjusted time or original string
+        }
+    }
+
+    // will check for the inventory for the product stock and if available then display the information on the UI
+    function displayStoreInformation(result) {
+
+        jQueryBopis('.hc-store-information').empty();
+        // TODO Handle it in a better way
+        // The content of error is not removed and appended to last error message
+        jQueryBopis('.hc-store-not-found').remove();
+        jQueryBopis('.hc-modal-content').append(jQueryBopis('<p class="hc-store-not-found"></p>'));
+    
+        //check for result count, result count contains the number of stores count in result
+        if (result.length > 0) {
+            result.map(async (store) => {
+                let $storeCard = jQueryBopis('<div id="hc-store-card"></div>');
+                let $storeInformationCard = jQueryBopis(`
+                <div id="hc-store-details">
+                    <div id="hc-details-column"><h4 class="hc-store-title">${store.locationName ? store.locationName : ''}</h4><p>${store.address ? store.address.addressLines[0] : ''}</p><p>${store.address.locality ? store.address.locality : ''} ${store.address.administrativeArea ? store.address.administrativeArea : ''} ${store.address.regionCode ? store.address.regionCode : ''}</p></div>
+                    <div id="hc-details-column"><p>In stock</p><p>${store.phone ? store.phone : ''}</p><p>Open Today: ${ store.regularHours ? tConvert(openData(store.regularHours).openTime) : ''} - ${store.regularHours ? tConvert(openData(store.regularHours).closeTime) : ''}</p></div>
+                </div>`);
+
+                let $pickUpButton = jQueryBopis('<button class="btn btn--secondary-accent hc-store-pick-up-button">Pick Up Here</button>');
+                $pickUpButton.on("click", updateCart.bind(null, store));
+
+                let $lineBreak = jQueryBopis('<hr/>')
+
+                $storeCard.append($storeInformationCard);
+                $storeCard.append($pickUpButton);
+                $storeCard.append($lineBreak);
+
+                jQueryBopis('.hc-store-information').append($storeCard);
+            })
+
+            //check whether the storeCard contains any children, if not then displaying error
+            if (!jQueryBopis('.hc-store-information').children().length) {
+                jQueryBopis('.hc-store-not-found').html('No stores found for current product');
+            }
+        } else {
+            jQueryBopis('.hc-store-not-found').append('No stores found for current product');
+        }
+
+        jQueryBopis('h4:empty').hide();
```

Comment:
Find in modal and then hide

Reply:
Sir, Improved code to use find method in the current modal